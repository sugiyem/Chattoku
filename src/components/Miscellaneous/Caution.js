import { Alert } from "react-native";
import React from "react";

const Caution = (text, onContinue) => {
  return Alert.alert(
    text,
    "This action is irreversible. Do you want to continue?",
    [{ text: "Cancel" }, { text: "Continue", onPress: onContinue }]
  );
};

export default Caution;
