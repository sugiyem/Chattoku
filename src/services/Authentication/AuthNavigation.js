export function redirectToForgotPasswordScreen(navigation) {
  navigation.replace("ForgotPassword");
}

export function redirectToLoginScreen(navigation) {
  navigation.replace("Login");
}

export function redirectToSignupScreen(navigation) {
  navigation.replace("Signup");
}
