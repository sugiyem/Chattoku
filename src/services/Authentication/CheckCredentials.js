export function isPasswordTooShort(password) {
  return password.length < 6;
}

// A str is considered a valid email if it is in the form of
// anystring@anystring.anystring
export function isValidEmail(str) {
  const regex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
  return regex.test(str);
}
