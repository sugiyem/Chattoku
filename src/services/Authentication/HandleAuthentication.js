import { Alert } from "react-native";
import { firebase } from "../Firebase/Config";
import { registerForPushNotificationsAsync } from "../Miscellaneous/HandleNotification";

export async function login(
  credentials,
  onSuccess,
  deviceToken = null,
  isVerified = false,
  app = firebase
) {
  await app
    .auth()
    .signInWithEmailAndPassword(credentials.email, credentials.password)
    .catch((e) => {
      console.log("Incorrect Credentials");
      Alert.alert("Incorrect Email / Password");
    });

  const currentUser = app.auth().currentUser;

  console.log(currentUser);

  if (currentUser === null) {
    return;
    // console.log("Incorrect Credentials");
    // Alert.alert("Incorrect Email / Password");
    // await firebase.auth().signOut();
  } else if (!isVerified && !currentUser.emailVerified) {
    console.log("email not verified");
    Alert.alert("Please Verify Your Email Before Logging In", "", [
      {
        text: "Resend verification email",
        onPress: () => currentUser.sendEmailVerification()
      },
      {
        text: "Cancel"
      }
    ]);

    signOut(app);
  } else {
    let token = null;

    if (deviceToken) {
      token = deviceToken;
    } else {
      token = await registerForPushNotificationsAsync();
    }

    if (token) {
      await app
        .firestore()
        .collection("users")
        .doc(currentUser.uid)
        .set({ notificationToken: token }, { merge: true });
    }

    onSuccess();
  }
}

export function signOut(app = firebase) {
  app.auth().signOut();
}

export async function signUp(credentials, onSuccess, app = firebase) {
  app
    .auth()
    .createUserWithEmailAndPassword(credentials.email, credentials.password)
    .then(async (response) => {
      response.user.sendEmailVerification();

      const currentUID = app.auth().currentUser.uid;

      await app.firestore().collection("users").doc(currentUID).set({
        username: credentials.username,
        bio: "",
        img: "",
        genres: [],
        friends: [],
        id: currentUID
      });

      Alert.alert(
        "Email sent.",
        "Please verify your account before signing in."
      );

      signOut(app);

      // console.log("finished");
    })
    .then(() => {
      onSuccess();
    })
    .catch((e) => {
      console.log(e);
      Alert.alert("email is already taken");
      console.log("email already taken");
    });
}

export async function sendForgotPasswordEmail(email, app = firebase) {
  await app
    .auth()
    .sendPasswordResetEmail(email)
    .then(() => {
      Alert.alert(
        "Email successfully sent",
        "An email to reset your password has been sent." +
          " Do check your spam folder"
      );
    })
    .catch((error) => {
      Alert.alert("Email is not registered");
    });
}
