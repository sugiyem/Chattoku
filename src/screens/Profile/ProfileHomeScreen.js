import React, { useEffect, useState } from "react";
import { Alert, Platform } from "react-native";
import { ListItem } from "react-native-elements";
import { useNavigation } from "@react-navigation/native";
import { firebase } from "../../services/Firebase/Config";
import { favoriteType } from "../../constants/Favorite";
import { deleteAccount } from "../../services/Authentication/HandleAuthentication";
import FetchFavoriteAnime from "../../services/Anime/FetchFavoriteAnime";
import FetchUserInfo from "../../services/Profile/FetchUserInfo";
import RenderFavorites from "../../components/Profile/RenderFavorites";
import styled from "styled-components/native";
import Loading from "../../components/Miscellaneous/Loading";
import Caution from "../../components/Miscellaneous/Caution";

const initialState = {
  username: "",
  bio: "",
  img: "",
  genres: []
};

const ProfileHomeScreen = () => {
  const [userInfo, setUserInfo] = useState(initialState);
  const [favoriteAnime, setFavoriteAnime] = useState([]);
  const [animeExpanded, setAnimeExpanded] = useState(false);
  const [genreExpanded, setGenreExpanded] = useState(false);
  const [isLoading, setIsLoading] = useState(false);

  const navigation = useNavigation();

  const datas = [
    {
      title: "Favorite Genre",
      render: ({ items }) => (
        <RenderFavorites
          type={favoriteType.GENRE}
          items={items}
          navigation={navigation}
        />
      ),
      data: userInfo.genres,
      isExpanded: genreExpanded,
      changeExpanded: setGenreExpanded
    },
    {
      title: "Favorite Anime",
      render: ({ items }) => (
        <RenderFavorites
          type={favoriteType.ANIME}
          items={items}
          navigation={navigation}
        />
      ),
      data: favoriteAnime,
      isExpanded: animeExpanded,
      changeExpanded: setAnimeExpanded
    }
  ];

  async function logOut() {
    try {
      navigation.replace("Login");
      firebase.auth().signOut();
    } catch (error) {
      Alert.alert(error.message);
    }
  }

  function onDelete() {
    Caution("This account will be deleted", async () => {
      setIsLoading(true);
      await deleteAccount(
        () => {
          navigation.replace("Login");
          Alert.alert("Account has been deleted");
        },
        (error) => Alert.alert("Error", error.message)
      ).finally(() => setIsLoading(false));
    });
  }

  useEffect(() => {
    if (isLoading) {
      return;
    }

    return FetchUserInfo({
      onSuccesfulFetch: (userInfo) => {
        setUserInfo(userInfo);
      },
      onFailure: (error) => {
        Alert.alert(error.message);
      }
    });
  }, [isLoading]);

  useEffect(() => {
    if (isLoading) {
      return;
    }

    return FetchFavoriteAnime({
      onSuccesfulFetch: (favorite) => {
        setFavoriteAnime(favorite);
      },
      onFailure: (error) => {
        Alert.alert(error.message);
      }
    });
  }, [isLoading]);

  return (
    <Loading isLoading={isLoading}>
      <Container contentContainerStyle={{ alignItems: "center" }}>
        <ContentContainer>
          {userInfo.img.length > 0 ? (
            <ProfilePicture source={{ uri: userInfo.img }} />
          ) : (
            <ProfilePicture
              source={require("../../assets/default-profile.png")}
            />
          )}

          <Username>{userInfo.username}</Username>
          <Bio>{userInfo.bio}</Bio>
          <ButtonGroup>
            <Button
              onPress={() => {
                navigation.navigate("PastPosts");
              }}
            >
              <ButtonText> See Recent Posts </ButtonText>
            </Button>
            <Button
              onPress={() => {
                navigation.navigate("EditProfile", { userInfo: userInfo });
              }}
            >
              <ButtonText>Edit Profile</ButtonText>
            </Button>
          </ButtonGroup>
          <ButtonGroup>
            <Button onPress={logOut}>
              <ButtonText>Logout</ButtonText>
            </Button>
            <Button onPress={onDelete}>
              <ButtonText>Delete Account</ButtonText>
            </Button>
          </ButtonGroup>
        </ContentContainer>

        <FavoriteStuffContainer>
          {datas.map((item, index) => (
            <ListItem.Accordion
              bottomDivider
              key={index}
              content={
                <ListItem.Content>
                  <ListItem.Title>
                    <Title>{item.title}</Title>
                  </ListItem.Title>
                </ListItem.Content>
              }
              isExpanded={item.isExpanded}
              onPress={() => item.changeExpanded(!item.isExpanded)}
            >
              {item.isExpanded && <item.render items={item.data} />}
            </ListItem.Accordion>
          ))}
        </FavoriteStuffContainer>
      </Container>
    </Loading>
  );
};

export default ProfileHomeScreen;

const Container = styled.ScrollView`
  background-color: darkcyan;
  padding: 5px;
  flex: 1;
`;

const ContentContainer = styled.View`
  align-items: center;
  justify-content: center;
  margin-top: 10px;
`;

const FavoriteStuffContainer = styled.View`
  align-self: stretch;
  border-top-width: 1px;
  border-top-color: black;
  margin: 10px;
  padding: 5px;
`;

const ProfilePicture = styled.Image`
  height: 150px;
  width: 150px;
  border-radius: 75px;
`;

const Username = styled.Text`
  font-family: ${Platform.OS === "ios" ? "Gill Sans" : "serif"};
  font-size: 20px;
  font-weight: bold;
  text-align=center;
  margin-vertical: 10px;
`;

const Bio = styled.Text`
  font-size: 15px;
  font-weight: 600;
  color: aquamarine;
  text-align: center;
  margin-bottom: 10px;
`;

const ButtonGroup = styled.View`
  flex-direction: row;
  justify-content: center;
  align-items: center;
  margin-bottom: 10px;
`;

const Button = styled.TouchableOpacity`
  border-color: navy;
  border-width: 2px;
  border-radius: 3px;
  background-color: white;
  padding-vertical: 8px;
  padding-horizontal: 12px;
  margin-horizontal: 5px;
`;

const ButtonText = styled.Text`
  color: #2e64e5;
`;

const Title = styled.Text`
  font-family: ${Platform.OS === "ios" ? "Gill Sans" : "serif"};
  font-size: 20px;
  font-weight: 600;
  text-decoration-line: underline;
`;
