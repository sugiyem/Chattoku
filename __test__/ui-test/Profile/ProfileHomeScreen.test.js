import React from "react";
import renderer from "react-test-renderer";
import { fireEvent, render } from "@testing-library/react-native";
import ProfileHomeScreen from "../../../src/screens/Profile/ProfileHomeScreen";
import Caution from "../../../src/components/Miscellaneous/Caution";
import { signOut } from "../../../src/services/Authentication/HandleAuthentication";

const mockNavigate = jest.fn();
const mockReplace = jest.fn();

jest.mock("../../../src/components/Miscellaneous/Caution");
jest.mock("@react-navigation/native", () => ({
  useNavigation: () => ({
    navigate: mockNavigate,
    replace: mockReplace
  })
}));
jest.mock("../../../src/services/Profile/FetchUserInfo");
jest.mock("../../../src/services/Anime/FetchFavoriteAnime");
jest.mock("../../../src/services/Authentication/HandleAuthentication");

describe("Test Profile Home Screen UI", () => {
  it("Renders correctly", () => {
    const tree = renderer.create(<ProfileHomeScreen />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("All interactable components are working as expected", () => {
    const { getByTestId } = render(<ProfileHomeScreen />);

    // Edit button is working
    fireEvent.press(getByTestId("editProfile"));
    expect(mockNavigate).toHaveBeenLastCalledWith(
      "EditProfile",
      expect.anything()
    );

    // Previous Post button is working
    fireEvent.press(getByTestId("pastPosts"));
    expect(mockNavigate).toHaveBeenLastCalledWith("PastPosts");

    // Logout button is working
    fireEvent.press(getByTestId("logOut"));
    expect(mockReplace).toHaveBeenCalledWith("Login");
    expect(signOut).toHaveBeenCalled();

    // Delete button is working
    fireEvent.press(getByTestId("delete"));
    expect(Caution).toHaveBeenCalledWith(
      "This account will be deleted",
      expect.anything()
    );
  });
});
