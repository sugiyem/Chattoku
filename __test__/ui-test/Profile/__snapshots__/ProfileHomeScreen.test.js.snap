// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Profile Home Screen UI Renders correctly 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    Object {
      "colors": Array [
        4287090411,
        4283256319,
      ],
      "endPoint": Array [
        1,
        1,
      ],
      "locations": undefined,
      "startPoint": Array [
        0,
        0,
      ],
    }
  }
  style={
    Object {
      "flex": 1,
      "padding": 5,
    }
  }
>
  <RCTScrollView
    style={
      Object {
        "flex": 1,
        "padding": 5,
      }
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginTop": 10,
            },
          ]
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "borderRadius": 75,
                "height": 150,
                "width": 150,
              },
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Gill Sans",
                "fontSize": 20,
                "fontWeight": "bold",
                "marginVertical": 10,
              },
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "aquamarine",
                "fontSize": 15,
                "fontWeight": "600",
                "marginBottom": 10,
                "textAlign": "center",
              },
            ]
          }
        >
          
        </Text>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": 10,
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#44d0fe",
                "borderColor": "navy",
                "borderRadius": 5,
                "borderWidth": 1,
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "marginHorizontal": 5,
                "opacity": 1,
                "paddingHorizontal": 12,
                "paddingVertical": 8,
              }
            }
            testID="pastPosts"
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "navy",
                    "fontSize": 16,
                    "fontWeight": "600",
                    "textAlign": "center",
                  },
                ]
              }
            >
               Recent Posts 
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#44d0fe",
                "borderColor": "navy",
                "borderRadius": 5,
                "borderWidth": 1,
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "marginHorizontal": 5,
                "opacity": 1,
                "paddingHorizontal": 12,
                "paddingVertical": 8,
              }
            }
            testID="editProfile"
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "navy",
                    "fontSize": 16,
                    "fontWeight": "600",
                    "textAlign": "center",
                  },
                ]
              }
            >
              Edit Profile
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": 10,
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#44d0fe",
                "borderColor": "navy",
                "borderRadius": 5,
                "borderWidth": 1,
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "marginHorizontal": 5,
                "opacity": 1,
                "paddingHorizontal": 12,
                "paddingVertical": 8,
              }
            }
            testID="logOut"
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "navy",
                    "fontSize": 16,
                    "fontWeight": "600",
                    "textAlign": "center",
                  },
                ]
              }
            >
              Logout
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#c10015",
                "borderColor": "whitesmoke",
                "borderRadius": 5,
                "borderWidth": 1,
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "marginHorizontal": 5,
                "opacity": 1,
                "paddingHorizontal": 12,
                "paddingVertical": 8,
              }
            }
            testID="delete"
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "whitesmoke",
                    "fontSize": 16,
                    "fontWeight": "600",
                    "textAlign": "center",
                  },
                ]
              }
            >
              Delete Account
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignSelf": "stretch",
              "borderTopColor": "black",
              "borderTopWidth": 1,
              "marginBottom": 10,
              "marginLeft": 10,
              "marginRight": 10,
              "marginTop": 10,
              "paddingBottom": 5,
              "paddingLeft": 5,
              "paddingRight": 5,
              "paddingTop": 5,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#87CEFA",
                "borderBottomWidth": 1,
                "borderColor": "#011F5B",
                "borderRadius": 10,
                "borderWidth": 1,
                "flexDirection": "row",
                "marginVertical": 5,
                "padding": 14,
              }
            }
            testID="padView"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "flex-start",
                    "flex": 1,
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
              theme={
                Object {
                  "colors": Object {
                    "black": "#242424",
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "grey": "rgba(0, 0, 0, 0.54)",
                        "primary": "#2196f3",
                        "searchBg": "#dcdce1",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "default": Object {
                        "error": "#ff3b30",
                        "grey": "#7d7d7d",
                        "primary": "#007aff",
                        "searchBg": "#dcdce1",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "grey": "#7d7d7d",
                        "primary": "#007aff",
                        "searchBg": "#dcdce1",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                      "web": Object {
                        "error": "#ff190c",
                        "grey": "#393e42",
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#ca71eb",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                    "white": "#ffffff",
                  },
                }
              }
            >
              <Text
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#242424",
                    "fontSize": 17,
                  }
                }
                testID="listItemTitle"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Gill Sans",
                        "fontSize": 20,
                        "fontWeight": "600",
                        "textDecorationLine": "underline",
                      },
                    ]
                  }
                >
                  Favorite Genre
                </Text>
              </Text>
            </View>
            <View
              style={
                Object {
                  "paddingLeft": 16,
                }
              }
            />
            <View
              collapsable={false}
              style={
                Object {
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            Object {
              "maxHeight": "0%",
              "opacity": 0,
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#87CEFA",
                "borderBottomWidth": 1,
                "borderColor": "#011F5B",
                "borderRadius": 10,
                "borderWidth": 1,
                "flexDirection": "row",
                "marginVertical": 5,
                "padding": 14,
              }
            }
            testID="padView"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "flex-start",
                    "flex": 1,
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
              theme={
                Object {
                  "colors": Object {
                    "black": "#242424",
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "grey": "rgba(0, 0, 0, 0.54)",
                        "primary": "#2196f3",
                        "searchBg": "#dcdce1",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "default": Object {
                        "error": "#ff3b30",
                        "grey": "#7d7d7d",
                        "primary": "#007aff",
                        "searchBg": "#dcdce1",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "grey": "#7d7d7d",
                        "primary": "#007aff",
                        "searchBg": "#dcdce1",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                      "web": Object {
                        "error": "#ff190c",
                        "grey": "#393e42",
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#ca71eb",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                    "white": "#ffffff",
                  },
                }
              }
            >
              <Text
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#242424",
                    "fontSize": 17,
                  }
                }
                testID="listItemTitle"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Gill Sans",
                        "fontSize": 20,
                        "fontWeight": "600",
                        "textDecorationLine": "underline",
                      },
                    ]
                  }
                >
                  Favorite Anime
                </Text>
              </Text>
            </View>
            <View
              style={
                Object {
                  "paddingLeft": 16,
                }
              }
            />
            <View
              collapsable={false}
              style={
                Object {
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            Object {
              "maxHeight": "0%",
              "opacity": 0,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</ViewManagerAdapter_ExpoLinearGradient>
`;
